Java/TelemetrySystem/src/main/java/tddmicroexercises/solidtelemetrysystem/TelemetryDiagnosticControlsImpl.java

```diff
@@ -0,0 +1,53 @@
+package tddmicroexercises.solidtelemetrysystem;
+
+import tddmicroexercises.telemetrysystem.TelemetryClient;
+
+public class TelemetryDiagnosticControlsImpl implements ITelemetryDiagnosticControls
+{
+
+    private final String DiagnosticChannelConnectionString = "*111#";
+
+    private final TelemetryClientImpl telemetryClient;
+
+    private final  TelemetryClientMessageSenderImpl telemetryClientMessageSender;
+    private String diagnosticInfo = "";
+
+    public TelemetryDiagnosticControlsImpl()
+        {
+            telemetryClient = new TelemetryClientImpl();
+            telemetryClientMessageSender = new TelemetryClientMessageSenderImpl();
+        }

```

Comment:
take telemetryClient and telmetryClientMessageSender from constructor parameters